components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -389,12 +391,18 @@ fn init_rng_data(expr: &mut Expr) -> Result<Option<RefCell<XorShiftRng>>> {
     if n == 1 {
         let arg = &mut children[0];
         match arg.get_tp() {
-            ExprType::ScalarFunc => return Ok(None),
             ExprType::ColumnRef => return Ok(Some(RefCell::new(get_rng(None)))),
```

Comment:
How do you distinguish it from the constant one below?

Reply:
I think the possible implement way:

1. if constant we cache the Rng.
2.if columRef we return Ok(None) and in rand_with_seed we use a default time seed to gen rand value.

I don't know if I am right. I have too many questions about how the cache metadata work and how to handle ColumRef. I am reading related code and articles now.

Reply:
Any similar fn example I can ref?

Reply:
@breeswish Do you have any suggestion?

Reply:
After the discussion, we think you can assume that it's ColumnRef, because TiDB should not push down the constant one as it's impossible to due with distributed request. And the `rand(NULL)` behavior of `TiDB` is also incorrect while `MySQL` always returns `0.155220427694936`.

All in all, we don't have to use metadata at all in `rand_with_seed`, we just create rng by the input, and give NULLs a default seed 0, who generates `0.155220427694936`.

Reply:
Sorry for the confusing explanation before 😢

Reply:
It's ok. Thank for your help!